datadog_lambda/metric.py

```diff
@@ -55,6 +56,20 @@ def lambda_metric(metric_name, value, timestamp=None, tags=None, force_async=Fal
     Note that if the extension is present, it will override the DD_FLUSH_TO_LOG value
     and always use the layer to send metrics to the extension
     """
+    if not metric_name or not isinstance(metric_name, str):
+        logger.warning(
+            "Ignoring metric submission. Invalid metric name: %s", metric_name
+        )
+        return
+
+    if not isinstance(value, numbers.Number):
+        logger.warning(
+            "Ignoring metric submission for metric '%s' because the value is not numeric: %r",
```

Comment:
Is there a chance that `lambda_metric("kittens", "1")` works? If so, then maybe we do something like

```py
try:
  int(value)
except:
  # log warning and return
  return
```